doc/sphinx/proof-engine/ltac.rst

```diff
@@ -1003,48 +1197,61 @@ Evaluation of a term can be performed with:
    :tacn:`compute`, :tacn:`simpl`, :tacn:`cbv`, :tacn:`lazy`, :tacn:`unfold`,
    :tacn:`fold`, :tacn:`pattern`.
 
-Recovering the type of a term
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+   :tacn:`eval` is a :token:`tactic_arg`.
+
+Getting the type of a term
+~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 .. tacn:: type of @term
 
    This tactic returns the type of :token:`term`.
 
+   :tacn:`type of` is a :token:`tactic_arg`.
+
 Manipulating untyped terms
 ~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. tacn:: uconstr : @term
+.. tacn:: uconstr : ( @term )
+
+   .. todo PR This is defined programmatically rather in the mlgs.  If
+      you do "Print Grammar tactic", it lists 28 others similar to uconstr
+      at the very end under "tactic_arg".  Not sure how to document these
+      or even if all of them should be documented.
```

Comment:
Why shouldn't we be documenting `tactic_arg`?  I believe all of them should be documented somehow.  Though perhaps for most of them we just want to say that `foo:( @foo )` causes the text inside the parens to be parsed as a `@foo`, and that this is true for the following syntax entries, and then list all of them?

Reply:
The special `tactic_arg`s (that don't follow this pattern) are: `fresh`, `type_term`, `numgoals`, and `constr_eval` (which itself is a different grammar entry not printed by `Print Grammar tactic`)
The ones that sort-of follow this are `glob_constr_with_bindings` (parses `constr_with_bindings`), `casted_constr` (parses `constr`), `lglob` (parses `lconstr`), `glob` (parses `constr`), `ltac` (parses `tactic_expr5`), `open_constr` (parses `lconstr`), `ipattern` (parses `simple_intropattern`), `constr` (parses `lconstr`), `uconstr` (parses `constr`)

Reply:
> Why shouldn't we be documenting tactic_arg?

I wanted to know whether some of them might be obsolete or deprecated before I wrote it up.  I see that the labels `lglob`, `lconstr`, `open_constr`, `constr` and `uconstr` all refer to the nonterminal `constr:lconstr`, which seems odd; AFAIK `lconstr` is for terms that may have `?ident` or `_` in them.  Both `casted_constr` and `glob` refer to `constr.constr`.

I wonder why this list is different from the `Tactic Notation` list.

Reply:
Also see https://github.com/coq/coq/issues/12302.

Reply:
> AFAIK `lconstr` is for terms that may have `?ident` or `_` in them

No, this is a misunderstanding of yours. `lconstr` is `term` and `constr` is `one_term`. The only relevant distinction between the two is the one that is explained in the [note about `one_term`](https://coq.github.io/doc/master/refman/language/core/basic.html#grammar-token-term). Both of them can be used as patterns. Cf. `Search` for an example which use `constr` / `one_term` for patterns.